### ISSUE TITLE
Post can’t be saved when all content is removed

### ISSUE THREAD

**Issue by gchtr** [WordPress/gutenberg#17346]
The `isEditedPostSaveable()` function introduced in https://github.com/WordPress/gutenberg/pull/5018 defines that a post can’t be saved when the content is empty:

https://github.com/WordPress/gutenberg/blob/4944315cac156c78313121657dff9d58ce8f4d3f/packages/editor/src/store/selectors.js#L471

This means that a post also can’t be saved when all content is removed from the post. It needs to have at least one block to be saveable.

In the following GIF, I load a post that has three empty paragraphs. When I remove the last remaining paragraph, the udpate button becomes locked.

![gutenberg-update-button-bug](https://user-images.githubusercontent.com/2084481/63917797-b2040580-ca3b-11e9-86c1-a5063fc67b72.gif)

I used empty paragraphs as an example, but it doesn’t matter whether it’s paragraph blocks or any other block.

**To reproduce**

1. Add a block to the post content.
2. Save the post and reload the browser window.
3. Remove all blocks.
4. Observe how the save button becomes locked.

**Expected behavior**

I don’t understand everything that’s involved here, with all the checks required for dirtyness etc. But I think the logic should be different so that when the content **becomes empty**, `isEditedPostEmpty()` should be ignored and I should be able to save the post.

**Desktop**

 - OS: macOS 10.14.6
 - Browser: Chrome
 - Version: 76

**Additional context**

- Gutenberg v6.4.0

**Comment by talldan** [comment]
@gchtr It'd be really useful to know what your use case is for saving an empty post.

**Comment by Soean** [comment]
For example if you just want to safe meta data for a post.

**Comment by gchtr** [comment]
I found that this only happens when **the post title is empty** and updated the issue accordingly.

Saving only meta data for a post is a use case that I have. The use case for an empty post title is that I generate it automatically in the `wp_insert_post_data` filter based on some of the meta values. The post title is not shown in the editor. I didn’t add `title` to `supports` in the post type configuration.

<img width="1334" alt="" src="https://user-images.githubusercontent.com/2084481/64598298-c939e000-d3b7-11e9-80b9-f5070d066ecc.png">

In this scenario, I can only save the post when I add at least one block to the content. Using the post content should be optional, though.

I opened this issue because of a more basic issue, where the editor allows users to reach a state where they suddenly can’t save their changes anymore, which I think is confusing.

@talldan If you think that this is an edge case that should be handled with a well-placed `unlockPostSaving`, then you can close this issue.

**Comment by skorasaurus** [comment]
Currently, even after manually invoking unlockPostSaving in the console of a post, I was still unable to save or publish a post without typing content into the Title. 

Our use case is that we'd like to link to articles that others have written about us and then provide links to those articles (as well as information about the article - Publisher of said article, and date it was published) in a dynamic block. 
I had tentatively mapped out our fields through ACF. 

I created a brief screencast that displays that even after invoking `wp.data.dispatch( 'core/editor' ).unlockPostSaving();` in the console in a post whose content and title is blank (with AC Fields included), the publish button is still not unlocked. 

The URL to view is at https://f002.backblazeb2.com/file/cpl-public/2020-04-09-gutenberg-bug-17346-withpostlock.mp4

I may workaround around this by using one of the custom fields as the post title. 

I understand this a bit of an edge case and for what it's worth, this behavior is possible in the classic editor (perhaps unintentionally?).

**Comment by rodrigo-arias** [comment]
I have a similar scenario in which the title is built with different full name fields, so the post type has no title support. My findings were the same as @skorasaurus.

**Comment by barryceelen** [comment]
Probably the along the same lines as @rodrigo-arias, we're also using post meta fields to populate the title with the name of a person. The content is optional, preventing us to save for a post type without title support. 

As mentioned in #27040 saving a post is also disabled if the user only updates the featured image so this issue does not only apply to use of meta boxes.

**Comment by anthonypero** [comment]
I have a similar use case. I want to auto-populate the title of a "Person" post-type with ACF fields, and the post type has lots of ACF fields and will only occasionally use the new Editor for biographies...but I want the OPTION to use the new editor for those biographies, rather than disable it entirely to make this work. The problem is that with the "Publish" button disabled, ACF can't perform the operation to add the fields to post_title in the first place, unless a block is added to the Editor.

**Comment by mediaformat** [comment]
Adding a use case similar to the ones already listed.

- We are generating post title based on post-meta (on save).
- We have blocks in the post content 
However because the block contents are populated via the Block Bindings API, we face this limitation.

I believe this PR https://github.com/WordPress/gutenberg/pull/45917, using `isEditedPostDirty( state ) )` would address the current limitation

**Comment by SirLouen** [comment]
== Reproduction Report
=== Description
✅ This report validates that the issue can be reproduced.

=== Environment
- WordPress: 6.9-alpha-60093-src
- PHP: 8.2.29
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.8.2-MariaDB-ubu2404 / Client: mysqlnd 8.2.29)
- Browser: Chrome 138.0.0.0
- OS: Windows 10/11
- Theme: Twenty Twenty-Four 1.3
- MU Plugins: None activated
- Plugins:
  * Gutenberg 21.3.0-rc.1
  * Test Reports 1.2.0

=== Testing Instructions
1. As provided in the OP 
2. I'm attaching a screencast with the steps executed.

=== Actual Results
1.  ✅ Issue occurs 

=== Additional Notes
- Basically we can save the post to a point that leaving it empty, doesn't provide the option to be saved (saving an empty post). The minimal state is at least with a title, or with some content (despite block is empty, at least an empty block)

- I've reread the comments, and I've found that the use case is to simply save metadata for such an empty post.

- @mediaformat you can submit and reference another PR if that one is stalling. [Please review first the testing report](https://github.com/WordPress/gutenberg/pull/45917#issuecomment-3132994882) for that PR to see what things have been done wrongly and improve them in case you could be thinking on submitting a new PR.

=== Supplemental Artifacts

Screencast: https://f003.backblazeb2.com/file/wordpress-videos/wp-videos/2025/07/gb17346.mp4

**Comment by mediaformat** [comment]
After further testing, our use case relates to inserting a pattern, via register post type - template, containing bound blocks.

**Comment by SirLouen** [comment]
> After further testing, our use case relates to inserting a pattern, via register post type - template, containing bound blocks.

If you want to submit a PR with a possible solution, go ahead, just ignore the current PR.

**Comment by SirLouen** [comment]
@westonruter suggested testing with `wp_insert_post_empty_content` filter

**Comment by SirLouen** [comment]
I've tested using the provided code in the PR #45917 and despite its working, I feel that it's not the right solution. Plainly unlocking the interface for the "dirty" state, brings a new bug (because without the filter, it's not going to be saveable and this introduces a massive UX bug). 

I think that the patch needs more work.

**Comment by mediaformat** [comment]
> despite its working, I feel that it's not the right solution.

@SirLouen agreed, the patch leaves the editor wide open to saving actually empty posts!



> [@westonruter](https://github.com/westonruter) suggested testing with `wp_insert_post_empty_content` filter

The problem with [wp_insert_post_empty_content](https://developer.wordpress.org/reference/hooks/wp_insert_post_empty_content/) is that it is geared to the actual saving (PHP)

Adding a screencast here to show my case where a post has a template of bound blocks.

https://github.com/user-attachments/assets/bc1596a1-4b25-43bd-9869-5c16657ec151
  
In this case what might be useful is an editor hook, or an additional test against post-meta. I'm still getting to know the internals, but `isEditedPostSaveable` might be where I'd place such a hook.

```
/**
 * Returns true if the post can be saved, or false otherwise. A post must
 * contain a title, an excerpt, or non-empty content to be valid for save.
 *
 * @param {Object} state Global application state.
 *
 * @return {boolean} Whether the post can be saved.
 */
export function isEditedPostSaveable( state ) {
	if ( isSavingPost( state ) ) {
		return false;
	}

	// TODO: Post should not be saveable if not dirty. Cannot be added here at
	// this time since posts where meta boxes are present can be saved even if
	// the post is not dirty. Currently this restriction is imposed at UI, but
	// should be moved here.
	//
	//  See: `isEditedPostPublishable` (includes `isEditedPostDirty` condition)
	//  See: <PostSavedState /> (`forceIsDirty` prop)
	//  See: <PostPublishButton /> (`forceIsDirty` prop)
	//  See: https://github.com/WordPress/gutenberg/pull/4184.

	return (
		!! getEditedPostAttribute( state, 'title' ) ||
		!! getEditedPostAttribute( state, 'excerpt' ) ||
		! isEditedPostEmpty( state ) ||
		Platform.OS === 'native'
	);
}
```

**Comment by mediaformat** [comment]
So there seems to be a genuine bug here, when considering postMeta.  

There is a recognition in `isEditedPostAutosaveable` that post meta is worth checking for:
https://github.com/WordPress/gutenberg/blob/a7ebbc00f5362dc0a0f835664a3348d8eee39ec7/packages/editor/src/store/selectors.js#L664-L669

However this check is not included in `isEditedPostSaveable` nor in `isEditedPostEmpty`.
